### PR TITLE
Add concurrency + timeout to Github Actions (GHA)

### PR DESCRIPTION
**Pull request by skoudoro**
This PR adds new attributes to GHA:

- Concurrency:  kill all the remaining jobs on a PR if there is a more recent commit
- Timeout: our test duration is approximately 40-45 min.  I added a hard limit to 90min in case something weird is happening. We should keep the timeout to 6h for the future doc generation matrix.

### CODE REVIEW

File: .github/workflows/test_template.yml [hunk 1]
```diff
@@ -30,6 +30,11 @@ on:
         type: string
         required: false
         default: '["pip", ]'
+      # timeout:
```
Comment (arokem) [inline .github/workflows/test_template.yml:33]: Do you mean for this to be commented out?

Comment (skoudoro) [reply .github/workflows/test_template.yml:33]: I wanted to make it a parameter of this template it but does not seem to work. This is a temporary comment until I understand what is going on

**Comment by skoudoro** [comment]
it seems it does not work. I will look deeper a bit later. Closing this PR for now.

**Comment by arokem** [comment]
BTW: Be careful with the tmate trick. Apparently GitHub considers this to be a TOS violation in some cases: https://github.com/mxschmitt/action-tmate/issues/104

**Comment by skoudoro** [comment]
> BTW: Be careful with the tmate trick. Apparently GitHub considers this to be a TOS violation in some cases: https://github.com/mxschmitt/action-tmate/issues/104

Ok, thank you for the info. Sometimes, this is the only way to understand what is going on and debug. 

I will remove this comment from another PR, it will avoid giving ideas to some people.

**Pull request closed without merging**